src/contracts/Vault721Adapter.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.24;
+
+import {IERC7496} from 'shipyard-core/src/dynamic-traits/interfaces/IERC7496.sol';
+import {IVault721} from '../interfaces/IVault721.sol';
+
+/**
+ * @notice IERC7496 events are never emitted since NFVState is tracked in Vault721
+ */
+contract Vault721Adapter is IERC7496 {
+  bytes32 public constant COLLATERAL = keccak256('COLLATERAL');
+  bytes32 public constant DEBT = keccak256('DEBT');
+
+  IVault721 public vault721;
+
+  error Disabled();
+  error UnknownTraitKeys();
+
+  constructor(IVault721 _vault721) {
+    vault721 = _vault721;
+  }
+
+  /**
+   * @dev get NFV trait
+   */
+  function getTraitValue(uint256 _tokenId, bytes32 _traitKey) public view returns (bytes32) {
+    (bytes32 _collateral, bytes32 _debt) = _getTraitValues(_tokenId);
+    if (_traitKey == COLLATERAL) return _collateral;
+    if (_traitKey == DEBT) return _debt;
+    else revert UnknownTraitKeys();
```

Comment:
I would prefer if this logic were in `_getTraitValue` (renamed from `_getTraitValues`)